tiledb/sm/enums/datatype.h

```diff
@@ -71,9 +73,22 @@ inline uint64_t datatype_size(Datatype type) {
       return sizeof(uint32_t);
     case Datatype::UINT64:
       return sizeof(uint64_t);
-    default:
-      return 0;
+    case Datatype::STRING_ASCII:
+      return sizeof(char);
+    case Datatype::STRING_UTF8:
+      return sizeof(char);
+    case Datatype::STRING_UTF16:
+      return sizeof(char);
+    case Datatype::STRING_UTF32:
+      return sizeof(char);
```

Comment:
I am not up-to-date on the unicode discussion that happened, but this seems wrong to me...

Reply:
These are both not right,  should be 16 and 32 bytes respectively.